.github/workflows/docker.yaml

```diff
@@ -229,7 +229,7 @@ jobs:
           gcloud auth configure-docker --quiet
       - name: Get canonical image name
         id: name
-        run: echo "canonical=$(echo ${{ needs.image.outputs.images }} | sed 's/ .*//')" >> "$GITHUB_OUTPUT"
+        run: echo "canonical=$(echo ${{ needs.image.outputs.images }} sed 's/ /\n/g' | awk '{cur=length($0); recs[cur] = recs[cur] $0 ORS; max=(cur>max?cur:max)} END{printf "%s", recs[max]}')" >> "$GITHUB_OUTPUT"
```

Comment:
This made me scratch my head a bit - I guess `recs[]` is an array indexed by the length of the string rather than by the position of the string in the list.
`awk '{cur=length($0); if (cur > max){longest = $0; max = cur;}} END {printf "%s", longest}'`
does the same thing without the array.